File: leetcode/0034-find-first-and-last-position-of-element-in-sorted-array/another-solution/solution.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> ans(2);
        ans[0] = -1;
        ans[1] = -1;
        
        int lo = findFirst(nums, target);
        ans[0] = lo < nums.size() && nums[lo] == target ? lo : -1;
        if (ans[0] == -1)
            return ans;
        
        ans[1] = findFirst(nums, target + 1) - 1;
        return ans;
    }
    
    int findFirst(vector<int>& nums, int target) {
        int lo = 0, hi = nums.size();
        
        while (lo < hi) {
            int mid = lo + (hi - lo) / 2;
            
            if (nums[mid] < target)
                lo = mid + 1;
            else
                hi = mid;
        }
        
        return lo;
    }
};

File: leetcode/0034-find-first-and-last-position-of-element-in-sorted-array/another-solution/solution.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        ans = list()
        ans.append(-1)
        ans.append(-1)
        
        lo = self.findFirst(nums, target)
        
        ans[0] = lo if lo < len(nums) and nums[lo] == target else -1
        
        if ans[0] == -1:
            return ans
        
        ans[1] = self.findFirst(nums, target + 1) - 1
        return ans
        
    
    def findFirst(self, nums: List[int], target: int):
        lo, hi = 0, len(nums)
        
        while lo < hi:
            mid = lo + (hi - lo) // 2
            
            if nums[mid] < target:
                lo = mid + 1
            else:
                hi = mid
        
        return lo